driver/compenv.ml

```diff
@@ -443,6 +454,39 @@ let read_one_param ppf position name v =
         let pass = Option.get (P.of_string v)  in
         Clflags.stop_after := Some pass
     end
+
+  | "save-ir-after" ->
+    if !native_code then begin
+      let module P = Clflags.Compiler_pass in
+      let passes = P.available_pass_names
+                     ~filter:P.can_save_ir_after
+                     ~native:!native_code in
+      begin match List.find_opt (String.equal v) passes with
+      | None ->
+          Printf.ksprintf (print_error ppf)
+            "bad value %s for option \"save-ir-after\" (expected one of: %s)"
+            v (String.concat ", " passes)
+      | Some v ->
+          let pass = Option.get (P.of_string v)  in
+          set_save_ir_after pass true
+      end
+    end
+
+   | "start-from" ->
+     let module P = Clflags.Compiler_pass in
+     let passes = P.available_pass_names
+                    ~filter:P.can_start_from
+                    ~native:!native_code in
+     begin match List.find_opt (String.equal v) passes with
+     | None ->
+         Printf.ksprintf (print_error ppf)
+           "bad value %s for option \"start-from\" (expected one of: %s)"
+           v (String.concat ", " passes)
+     | Some v ->
+         let pass = Option.get (P.of_string v) in
+         Clflags.start_from := Some pass
+     end
```

Comment:
The code looks quite similar between `save-ir-after` and `start-from`, we could factor it with something like:

```ocaml
let pass = decode_compiler_pass ~option_name:name ~filter:... in
...
```

Reply:
done